### ISSUE TITLE
re-entering your body resets the respawn timer

### ISSUE THREAD

**Issue by Kukutis96513** [Sector-Umbra/Sector-Umbra#140]
## Description

Re-entering your body resets the respawn timer.
**Reproduction**

have a skill issue and perish
wait abit
re-enter your body
exit your body
respawn timer resets
**Screenshots**

**Additional context**

i think the respawn timer might be tied to the ghost? and re-entering your body destroys your ghost, though i'm unsure about it.

**Comment by dffdff2423** [comment]
Yes, it is related to the time you are a ghost. The only way around that would be to attach the death time somehow to the player instead of the entity they control
